src/main/java/uk/gov/justice/hmpps/prison/service/AppointmentsService.java

```diff
@@ -137,6 +137,18 @@ public ScheduledEvent getBookingAppointment(Long appointmentId) {
         return getScheduledEventOrThrowEntityNotFound(appointmentId);
     }
 
+    @Transactional
+    @PreAuthorize("hasAnyRole('GLOBAL_APPOINTMENT')")
+    public void deleteBookingAppointments(List<Long> appointmentIds) {
+        appointmentIds.forEach(appointmentId -> bookingRepository
```

Comment:
This is quite inefficient - I think it can all be done in bulk. But you would need to detect which id's are present first so the the telemetry calls are accurate, which could be done in a single query rather than getting the whole object for each appt.

I don't think it makes a massive difference though - it's up to you if you want to change it.

Reply:
I did consider just deleting them in a batch but realised we still need to fetch each one to gather info for the tracking event.